Compiled from "ErrorCollector.java"
public class org.junit.rules.ErrorCollector extends org.junit.rules.Verifier {
  private java.util.List<java.lang.Throwable> errors;

  public org.junit.rules.ErrorCollector();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method org/junit/rules/Verifier."<init>":()V
       4: aload_0
       5: new           #2                  // class java/util/ArrayList
       8: dup
       9: invokespecial #3                  // Method java/util/ArrayList."<init>":()V
      12: putfield      #4                  // Field errors:Ljava/util/List;
      15: return

  protected void verify() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #4                  // Field errors:Ljava/util/List;
       4: invokestatic  #5                  // Method org/junit/runners/model/MultipleFailureException.assertEmpty:(Ljava/util/List;)V
       7: return

  public void addError(java.lang.Throwable);
    Code:
       0: aload_1
       1: ifnonnull     14
       4: new           #6                  // class java/lang/NullPointerException
       7: dup
       8: ldc           #7                  // String Error cannot be null
      10: invokespecial #8                  // Method java/lang/NullPointerException."<init>":(Ljava/lang/String;)V
      13: athrow
      14: aload_1
      15: instanceof    #9                  // class org/junit/internal/AssumptionViolatedException
      18: ifeq          53
      21: new           #10                 // class java/lang/AssertionError
      24: dup
      25: aload_1
      26: invokevirtual #11                 // Method java/lang/Throwable.getMessage:()Ljava/lang/String;
      29: invokespecial #12                 // Method java/lang/AssertionError."<init>":(Ljava/lang/Object;)V
      32: astore_2
      33: aload_2
      34: aload_1
      35: invokevirtual #13                 // Method java/lang/AssertionError.initCause:(Ljava/lang/Throwable;)Ljava/lang/Throwable;
      38: pop
      39: aload_0
      40: getfield      #4                  // Field errors:Ljava/util/List;
      43: aload_2
      44: invokeinterface #14,  2           // InterfaceMethod java/util/List.add:(Ljava/lang/Object;)Z
      49: pop
      50: goto          64
      53: aload_0
      54: getfield      #4                  // Field errors:Ljava/util/List;
      57: aload_1
      58: invokeinterface #14,  2           // InterfaceMethod java/util/List.add:(Ljava/lang/Object;)Z
      63: pop
      64: return

  public <T> void checkThat(T, org.hamcrest.Matcher<T>);
    Code:
       0: aload_0
       1: ldc           #15                 // String
       3: aload_1
       4: aload_2
       5: invokevirtual #16                 // Method checkThat:(Ljava/lang/String;Ljava/lang/Object;Lorg/hamcrest/Matcher;)V
       8: return

  public <T> void checkThat(java.lang.String, T, org.hamcrest.Matcher<T>);
    Code:
       0: aload_0
       1: new           #17                 // class org/junit/rules/ErrorCollector$1
       4: dup
       5: aload_0
       6: aload_1
       7: aload_2
       8: aload_3
       9: invokespecial #18                 // Method org/junit/rules/ErrorCollector$1."<init>":(Lorg/junit/rules/ErrorCollector;Ljava/lang/String;Ljava/lang/Object;Lorg/hamcrest/Matcher;)V
      12: invokevirtual #19                 // Method checkSucceeds:(Ljava/util/concurrent/Callable;)Ljava/lang/Object;
      15: pop
      16: return

  public <T> T checkSucceeds(java.util.concurrent.Callable<T>);
    Code:
       0: aload_1
       1: invokeinterface #20,  1           // InterfaceMethod java/util/concurrent/Callable.call:()Ljava/lang/Object;
       6: areturn
       7: astore_2
       8: new           #10                 // class java/lang/AssertionError
      11: dup
      12: ldc           #21                 // String Callable threw AssumptionViolatedException
      14: invokespecial #12                 // Method java/lang/AssertionError."<init>":(Ljava/lang/Object;)V
      17: astore_3
      18: aload_3
      19: aload_2
      20: invokevirtual #13                 // Method java/lang/AssertionError.initCause:(Ljava/lang/Throwable;)Ljava/lang/Throwable;
      23: pop
      24: aload_0
      25: aload_3
      26: invokevirtual #22                 // Method addError:(Ljava/lang/Throwable;)V
      29: aconst_null
      30: areturn
      31: astore_2
      32: aload_0
      33: aload_2
      34: invokevirtual #22                 // Method addError:(Ljava/lang/Throwable;)V
      37: aconst_null
      38: areturn
    Exception table:
       from    to  target type
           0     6     7   Class org/junit/internal/AssumptionViolatedException
           0     6    31   Class java/lang/Throwable

  public void checkThrows(java.lang.Class<? extends java.lang.Throwable>, org.junit.function.ThrowingRunnable);
    Code:
       0: aload_1
       1: aload_2
       2: invokestatic  #24                 // Method org/junit/Assert.assertThrows:(Ljava/lang/Class;Lorg/junit/function/ThrowingRunnable;)Ljava/lang/Throwable;
       5: pop
       6: goto          15
       9: astore_3
      10: aload_0
      11: aload_3
      12: invokevirtual #22                 // Method addError:(Ljava/lang/Throwable;)V
      15: return
    Exception table:
       from    to  target type
           0     6     9   Class java/lang/AssertionError
}
